src/components/CreateVideo.js

```diff
@@ -1,32 +1,90 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { connect } from "react-redux";
 import { withRouter } from "react-router-dom";
+import { axiosWithAuth } from "../utils/axiosWithAuth";
+
 // Need to create action for posting video
-// import { postVideo } from "../store/actions";
+import { postVideo } from "../store/actions";
 import style from "styled-components";
 
-import Searchbar from "./Searchbar"
+
+const API_URL = "https://cors-anywhere.herokuapp.com/https://api.deezer.com/search?q=";
 
 const CreateVideo = props => {
-  const [video, setVideo] = useState({
-    title: "",
-    song: "",
-    artist: ""
+  // const [video, setVideo] = useState({
+  //   title: "",
+  //   song: "",
+  //   artist: ""
+  // });
+
+  // const handleChanges = event => {
+  //   setVideo({ ...video, [event.target.name]: event.target.value });
+  //   console.log(event.target.name);
+  // };
+
+  const [query, setQuery] = useState("");
+
+  const [results, setResults] = useState([]);
+
+  const [videoTitle, setVideoTitle] = useState("");
+
+  const [selectedSong, setSelectedSong] = useState({
+    title_short: "",
+    preview: "",
+    artist: "",
+    deezer_id: "",
+    location: "youtube.com/video",
+    video_title: "",
+    user_id: localStorage.getItem("user_id"),
   });
 
-  const handleChanges = event => {
-    setVideo({ ...video, [event.target.name]: event.target.value });
-    console.log(event.target.name);
+  const fullQuery = `${API_URL}${query}`;
+
+  const getInfo = () => {
+    console.log(fullQuery);
+    axiosWithAuth()
+      .get(`${fullQuery}&limit=7`)
+      .then((res) => {
+        console.log("res", res);
+        setResults(res.data.data);
+      })
+      .catch((err) => {
+        console.log("unable to suggest artist and/or song", err);
+      });
+  };
+
+  useEffect(() => {
+    if (query.length >= 2) getInfo();
+  }, [query]);
+
+  const handleSongChange = (event) => {
+    setQuery(event.target.value);
+  };
+
+  const handleTitleChange = (event) => {
+    setVideoTitle({ [event.target.name]: event.target.value });
+  };
+
+  const handleClickSong = (event) => {
+    const songItem = event.target.getAttribute("data-index");
+    setSelectedSong({
+      ...selectedSong,
+      title_short: results[songItem].title_short,
+      preview: results[songItem].preview,
+      artist: results[songItem].artist.name,
+      deezer_id: results[songItem].id,
+      video_title: videoTitle.title
```

Comment:
This is compact and legible.🙌🏻 Should be easy to 🧪unit test🧪 written this way. 
🐰